downgrade

```diff
@@ -103,8 +113,8 @@ main() {
   for term; do
     candidates=( $(search_packages "$term" | sort_packages) )
 
-    if present_packages "${candidates[@]}"; then
-      if choice="$(read_selection "${candidates[@]}")"; then
+    if check_packages || test "${#candidates[@]}" -gt 0; then
+      if check_selection || choice="${candidates[0]}"; then
```

Comment:
I see. I think it's confusing to hide the "does thing if there's more than one option" behind these `check_` function. WDYT of writing this something like this:

```bash
if (( "${#candidates[@]" == 1 )); then
  choice=${candidates[0]}
else
  # do what it does now with present/read-selection/etc
  choice=...
end

# proceed, operating on `$choice`
```

It might require a little shuffling of the loop's `continue` and setting of `exit_code`, but I think it'd be worthwhile, and hopefully there are tests to ensure you don't break it.